CONTRIBUTING.md

```diff
@@ -58,6 +58,54 @@ this will also help us to understand it better ourselves.
 always have more work to do than time to do it. If you can write some code
 then that will speed the process along.
 
+### Using Feature Flags
+
+Feature flags allow new features to be tested easily and strips them out of
+production builds automatically.
+
+1. Add your new feature flag to the [config/features.json](https://github.com/emberjs/data/blob/master/config/features.json) file
+
+  ```js
+  {
+    "ds-boolean-transform-allow-null": null,
+    "ds-mynew-feature": null
+  }
+  ```
+
+  Give it a default of `null` so it will not be used in production builds.
+
+2. Import `isEnabled` from `ember-data/-private/features`, wrapping any new
+code with your feature:
+
+  ```js
+  import isEnabled from 'ember-data/-private/features';
+
+  if (isEnabled('ds-mynew-feature')) {
+    // ... any additional code
+  } else {
+    // ... any previous code that may have been overwritten
+  }
+  ```
+
+3. Similarly, you will want to wrap any new or edited tests with the same feature flag
+
+  ```js
+  import isEnabled from 'ember-data/-private/features';
+
+  if (isEnabled('ds-mynew-feature')) {
+    test('test for new feature', function(assert) {
+      // ...
+    })
+  }
+  ```
+
+  This will allow the test suite to run as normal.
+
+4. Running tests with all feature flags enabled is possible via `ember test --environment=test-optional-features`
```

Comment:
Can you add a note that this is also possible in the browser via the `Enable Opt Features` checkbox. This is handy if you need to debug stuff which is behind a feature flag.